src/redux/books/booksSlice.js

```diff
@@ -1,46 +1,98 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
 
+const appId = 'q8FdluzHjZBBWXH5Y85r';
 const initialState = {
-  books: [
-    {
-      item_id: 'item1',
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item2',
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item3',
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    },
-  ],
+  books: [],
+  isLoading: false,
+  error: false,
 };
+export const getBooks = createAsyncThunk('books/getBooks', () => axios.get(`https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/${appId}/books`)
+  .then((response) => response.data));
+export const addBook = createAsyncThunk('books/addBook', async (book) => {
+  const result = await axios.post(`https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/${appId}/books`, book);
+  return result.data;
+});
+export const removeBook = createAsyncThunk('books/removeBook', async (id) => {
+  await axios.delete(`https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/${appId}/books/${id}`);
+  return id;
+});
 
 const booksSlice = createSlice({
   name: 'books',
   initialState,
   reducers: {
-    addBook: (state, action) => {
-      state.books.push(action.payload);
-    },
-    removeBook: (state, action) => {
-      const bookId = action.payload;
-      const filteredBooks = state.books.filter((book) => book.item_id !== bookId);
+    addNewBook: (state, action) => {
+      const bookItem = action.payload;
+      const book = {
+        item_id: bookItem.item_id,
+        title: bookItem.title,
+        author: bookItem.author,
+        category: bookItem.category,
+      };
```

Comment:
[OPTIONAL] Well done overall 💯 but I'd suggest here using object destructuring: The code currently extracts the `item_id`, `title`, `author`, and `category` properties from `bookItem` using dot notation. You could use object destructuring to extract these properties more cleanly, like this:
```
const { item_id, title, author, category } = bookItem;
const book = { item_id, title, author, category };
```